### PR TITLE
♻️Mypy: webserver2

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

Completes the webserver mypy errors... (sic)

## Related issue/s
- relates to https://github.com/ITISFoundation/osparc-simcore/issues/6115

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/payments/_events.py [hunk 1]
```diff
@@ -26,6 +26,9 @@ async def validate_prices_in_product_settings_on_startup(app: web.Application):
                     amount_usd=payment_settings.PAYMENTS_AUTORECHARGE_DEFAULT_TOP_UP_AMOUNT,
                     min_payment_amount_usd=product.min_payment_amount_usd,
                 )
+            assert (
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/payments/_events.py:29]: `# nosec` has to aligned with `assert`

**Review by pcrespov — APPROVED** [review]
thx

File: packages/models-library/src/models_library/basic_types.py [hunk 2]
```diff
@@ -83,11 +83,31 @@ class UUIDStr(ConstrainedStr):
 
 # non-empty bounded string used as identifier
 # e.g. "123" or "name_123" or "fa327c73-52d8-462a-9267-84eeaf0f90e3" but NOT ""
+_ELIDE_CHAR: Final[str] = "..."
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/models-library/src/models_library/basic_types.py:86]: good one

File: services/web/server/src/simcore_service_webserver/groups/_db.py [hunk 3]
```diff
@@ -51,14 +51,16 @@ async def _get_user_group(
...
     return group
 
 
 async def get_user_from_email(conn: SAConnection, email: str) -> RowProxy:
     result = await conn.execute(sa.select(users).where(users.c.email == email))
-    user: RowProxy = await result.fetchone()
+    user = await result.fetchone()
     if not user:
         raise UserNotFoundError(email=email)
+    assert isinstance(user, RowProxy)  # nosec
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/groups/_db.py:63]: I rather dislike annotate as this usually will get wrong when the signature of a function changes and might go invisible.
this one should make mypy fail and probably some test as well if that is tested.

File: services/web/server/src/simcore_service_webserver/payments/_events.py (hunk 1 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/payments/_events.py:29]: true that.. why is not complaining?

Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/payments/_events.py:29]: yep weird that this does not appera

**Review by GitHK — APPROVED** [review]
thanks for completing this 👍
Found some minor stuff.

**Pull request merged**